core/src/avm2/object/socket_object.rs

```diff
@@ -24,6 +24,8 @@ pub fn socket_allocator<'gc>(
             // Default endianness is Big.
             endian: Cell::new(Endian::Big),
             object_encoding: Cell::new(ObjectEncoding::Amf3),
+            // Default is 20_000 milliseconds or 20 seconds.
+            timeout: Cell::new(20000),
```

Comment:
For technical correctness with Flash, the timeout is set when instantiating the Socket, not when creating the object.  So the "default" should be `0`, and the constructor should do this:
```
        public function Socket(host:String = null, port:int = 0) {
            this.timeout = 20000; <-- timeout gets set here
            if (host != null) {
                this.connect(host, port);
            }
        }
```

Reply:
Done.